### PR TITLE
[KOGITO-4554] Error boundary event implementation

### PR DESCRIPTION
**Pull request by tiagodolphine**
This includes the missing Boundary Error Event implementation in Kogito.
The processes used to test and included in the integration tests module in codegen were:

- Boundary Error on Task

image

- Boundary Error on Sub-Process

image

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-build/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by tiagodolphine** [comment]
jenkins retest this

**Comment by tiagodolphine** [comment]
jenkins retest this

**Comment by tiagodolphine** [comment]
jenkins retest this

**Review by cristianonicolai — APPROVED** [review]
@tiagodolphine just a minor comment to review sonar where applicable

**Comment by tiagodolphine** [comment]
@cristianonicolai sonar is happy now.
@MarianMacik would you like to take a look at this one or are we ok to merge?
Thanks

**Comment by MarianMacik** [comment]
@tiagodolphine Looking into it.

File: kogito-codegen-modules/kogito-codegen-integration-tests/src/test/resources/error/BoundaryErrorSubProcess.bpmn2 [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        </bpmn2:extensionElements>
+      </bpmn2:sequenceFlow>
+      <bpmn2:endEvent id="_DD5046D6-4A18-4EF3-B3B4-7641DF6889AC" name="error1">
+        <bpmn2:extensionElements>
+          <drools:metaData name="elementname">
+            <drools:metaValue><![CDATA[error1]]></drools:metaValue>
+          </drools:metaData>
+        </bpmn2:extensionElements>
+        <bpmn2:incoming>_0DB71D4C-4929-49B1-88DE-3BE7D82B905C</bpmn2:incoming>
+        <bpmn2:errorEventDefinition drools:erefname="error2" errorRef="error2"/>
```
Comment (MarianMacik) [inline kogito-codegen-modules/kogito-codegen-integration-tests/src/test/resources/error/BoundaryErrorSubProcess.bpmn2:134]: endEvent is called error1 but returns error2. Is this deliberate to confuse the enemy? :)

**Review by MarianMacik — APPROVED** [review]
One minor thing with the test process definition. Otherwise looks good.

File: kogito-codegen-modules/kogito-codegen-integration-tests/src/test/resources/error/BoundaryErrorSubProcess.bpmn2 (hunk 1 above)
Comment (tiagodolphine) [reply kogito-codegen-modules/kogito-codegen-integration-tests/src/test/resources/error/BoundaryErrorSubProcess.bpmn2:134]: it was done on purpose, joking :)  just fixed it, thanks, great catch.

**Comment by tiagodolphine** [comment]
jenkins retest this

**Comment by cristianonicolai** [comment]
Jenkins rerun apps tests

**Comment by cristianonicolai** [comment]
Jenkins rerun runtimes tests

**Comment by cristianonicolai** [comment]
Jenkins rerun runtimes tests

**Comment by tiagodolphine** [comment]
Jenkins rerun runtimes tests

**Pull request merged**